patroni/dcs/__init__.py

```diff
@@ -248,7 +248,7 @@ def checkpoint_after_promote(self):
         if version:
             try:
                 if tuple(map(int, version.split('.'))) >= (1, 5, 6):
-                    return bool(self.member.data.get('checkpoint_after_promote'))
+                    return self.member.data['role'] == 'master' and 'checkpoint_after_promote' not in self.member.data
```

Comment:
1. does  the flag `checkpoint_after_promote` mean everywhere in the codebase that the checkpoint in question is happening right now ?
2. can `self.member.data`  contain `{checkpoint_after_promote : False}`  ?

Reply:
> can self.member.data contain {checkpoint_after_promote : False} ?

In fact, it will contain only this value. When checkpoint already has happened checkpoint_after_promote will disappear from the json.

Basically replicas only need to know the fact that checkpoint after promotion on the current primary didn't happen yet in order to postpone pg_rewind.